SU2_CFD/src/numerics/CNumerics.cpp

```diff
@@ -109,6 +103,72 @@ CNumerics::CNumerics(unsigned short val_nDim, unsigned short val_nVar,
 
   Dissipation_ij = 1.0;
 
+  //if ((config->GetKind_Solver() == NEMO_EULER)                 ||
+  //    (config->GetKind_Solver() == NEMO_NAVIER_STOKES) ) { //         ||
+      //(config->GetKind_Solver() == DISC_ADJ_NEMO_EULER)        ||
+      //(config->GetKind_Solver() == DISC_ADJ_NEMO_NAVIER_STOKES) ) 
+  if (config->GetNEMOProblem()){
+
+    nSpecies = nVar - nDim - 2;
+
+    hs     = new su2double[nSpecies];
+    Cvtr   = new su2double[nSpecies];
```

Comment:
To avoid bloating the base CNumerics I recommend moving some of these member variables and the logic to allocate them to an "intermediate" parent class between CNumerics and the NEMO classes.
You can keep the members that need to be set from the outside (to avoid inefficient virtual "setter" functions) but the "working" variables can be moved to another class without hurting performance.
You can see for example what Ruben did with the Radiation classes.

Reply:
Take the above as optional, but please do the following:
Delete the commented part,
Delete the variables in this block in the destructor, currently it seems they are all being leaked (consider running valgrind to find other such issues).